cmd/builder/command.go

```diff
@@ -15,6 +16,15 @@ type Command struct {
 	args    []string
 }
 
+// Executor holds available methods for command.
+type Executor interface {
+	AppendArgs(args ...string)
+	Interactive() error
+	String() string
+	Exec() (string, error)
```

Comment:
This note should be further discussed and dealt with down the road (next PR maybe) but here it goes - I do think we should split the logic of command builder and runner.

Two different tasks, which could make us have in the end a `Builder` interface and `DefaultBuilder` struct for the commands building, and then a `Runner` `DefaultRun` which encapsulates `shell.Exec/shell.Interactive` etc.